src/activemasternode.cpp

```diff
@@ -190,48 +187,56 @@ void CActiveMasternode::ManageStateInitial()
         eType = MASTERNODE_LOCAL;
     }
 
-    LogPrint("masternode", "CActiveMasternode::ManageStateInitial -- End status = %s, type = %s, pinger enabled = %d\n", GetStatus(), GetType(), fPingerEnabled);
+    LogPrint("masternode", "CActiveMasternode::ManageStateInitial -- End status = %s, type = %s, pinger enabled = %d\n", GetStatus(), GetTypeString(), fPingerEnabled);
 }
 
 void CActiveMasternode::ManageStateRemote()
 {
     LogPrint("masternode", "CActiveMasternode::ManageStateRemote -- Start status = %s, type = %s, pinger enabled = %d, pubKeyMasternode.GetID() = %s\n", 
-             GetStatus(),
-             GetType(),
-             fPingerEnabled,
-             pubKeyMasternode.GetID().ToString());
+             GetStatus(), fPingerEnabled, GetTypeString(), pubKeyMasternode.GetID().ToString());
 
     mnodeman.CheckMasternode(pubKeyMasternode);
-    masternode_info_t infoMn = mnodeman.GetMasternodeInfo(pubKeyMasternode);
-    if(infoMn.fInfoValid) {
-        if(infoMn.nProtocolVersion != PROTOCOL_VERSION) {
+    CMasternode mn;
```

Comment:
I needed `mn.GetStateString()` for a more precise error message, shouldn't add a lot of overhead here imo, `ManageStateRemote` is called initially and then only every `MASTERNODE_MIN_MNP_SECONDS`.